release-0.4.md

```diff
@@ -0,0 +1,80 @@
+- `contour.heptio.com/per-try-timeout` to control the request timeout _per attempt_.
+For more information please consult the [annotation documentation][4].
+Thanks @cmaloney.
+Fixes #164 and #221.
+
+### Ingress class now configurable
+
+By default Contour responds to the ingress class annotation `kubernetes.io/ingress.class: contour` where present.
+However, if required while migrating from another ingress controller, you can pass the flag `--ingress-class-name` to adjust the `ingress.class` name that Contour will respond to.
+Thanks @Nikkau.
+Fixes #255
+
+### TLS 1.1 is now mandatory
+
+Following the [advice of the PCI Security Standards Council][5] Contour 0.4 will configure Envoy to require TLS 1.1 or later.
+Thanks @sevein
+Fixes #185
+
+### Certificate information is now sent in-line in the gRPC message
+
+Due to a limitation in Envoy Contour 0.3 exchanged certificate data with Envoy via a shared file system.
+This limitation has been addressed and Contour 0.4 transmits certificate data directly to Envoy in gRPC API response messages. 
+Thanks @sevein
+Fixes #158
+
+### Contour and Envoy can now live in separate pods
+
+Although not currently utilised, Contour and Envoy can now exist in separate pods.
+This is acomplished by the `--xds-address` and `--xds-port` flags passed to both `contour bootstrap` and `contour serve` which can be used to deploy Contour as a ReplicaSet which Envoy is a Daemonset. Thanks to @sevein. Fixes #165
+
+### `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation now applies on a per Ingress basis
+
+Prior to Contour 0.4 if _any_ Ingress mentioned a virtual host used the `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation, then all routes for that virtual host would be 301 upgraded.
+In Contour 0.4 this behaviour is applied per Ingress, that is, all routes in an Ingress object.
+This permits so-called _split_ Ingress setups where some routes on a virtual host will be 301 ugpraded, and others not. 
+Fixes #250.
+
+### Other bug fixes and improvements in this release
+
+- The Contour docker image no longer bundles the `ocid` and `gcp` authentication plugins as they are not required for `-incluster` deployments.
+- Daemonset or Deployment examples now specify the `--v2-config-only` flag to instruct Envoy to not fall back to parsing an invalid configuration file as the deprecated v1 JSON format. This should aid debugging `contour bootstrap` issues. Thanks @cmaloney
+- Updated to the latest `envoyproxy/go-control-plane` library. Thanks @vaamarnath. Fixes #225
+- Contour has switched to sirupsen/logrus as its logging library. Fixes #162.
+- Our [troubleshooting][3] documentation now includes a simple way to access Envoy's Admin interface which is useful for examining the state of its route and cluster tables.
+- Various issues related to updating Services and Ingresses in place via `kubectl edit` have been by the introduction of a new caching layer in the translator package. 
```

Comment:
Issues related to updating Service and Ingress objects in place with `kubectl edit` are fixed with a new caching layer in the translator package.